diff --git a/test/ruby/classic_rails/application_spawner_spec.rb b/test/ruby/classic_rails/application_spawner_spec.rb
index 20355b1..415456d 100644
--- a/test/ruby/classic_rails/application_spawner_spec.rb
+++ b/test/ruby/classic_rails/application_spawner_spec.rb
@@ -37,8 +37,8 @@ describe ClassicRails::ApplicationSpawner do
 			it_should_behave_like "a spawner"
 			it_should_behave_like "a spawner that does not preload app code"
 			it_should_behave_like "a Rails spawner"
-			include_shared_example_group "a Rails app that lacks RAILS_GEM_VERSION"
-			include_shared_example_group "analytics logging extensions for Rails"
+			it_should_behave_like "a Rails app that lacks RAILS_GEM_VERSION"
+			it_should_behave_like "analytics logging extensions for Rails"
 		end
 	end
 	
@@ -80,8 +80,8 @@ describe ClassicRails::ApplicationSpawner do
 			it_should_behave_like "a spawner"
 			it_should_behave_like "a spawner that preloads app code"
 			it_should_behave_like "a Rails spawner"
-			include_shared_example_group "a Rails app that lacks RAILS_GEM_VERSION"
-			include_shared_example_group "analytics logging extensions for Rails"
+			it_should_behave_like "a Rails app that lacks RAILS_GEM_VERSION"
+			it_should_behave_like "analytics logging extensions for Rails"
 		end
 	end
 end
diff --git a/test/ruby/rack/application_spawner_spec.rb b/test/ruby/rack/application_spawner_spec.rb
index 7da120a..a8040ab 100644
--- a/test/ruby/rack/application_spawner_spec.rb
+++ b/test/ruby/rack/application_spawner_spec.rb
@@ -48,7 +48,7 @@ describe Rack::ApplicationSpawner do
 				return register_app(app)
 			end
 			
-			include_shared_example_group "analytics logging extensions for Rails"
+			it_should_behave_like "analytics logging extensions for Rails"
 		end
 	end
 	
@@ -108,7 +108,7 @@ describe Rack::ApplicationSpawner do
 				return register_app(app)
 			end
 			
-			include_shared_example_group "analytics logging extensions for Rails"
+			it_should_behave_like "analytics logging extensions for Rails"
 		end
 	end
 end
diff --git a/test/ruby/spawn_manager_spec.rb b/test/ruby/spawn_manager_spec.rb
index 6140fc3..987a020 100644
--- a/test/ruby/spawn_manager_spec.rb
+++ b/test/ruby/spawn_manager_spec.rb
@@ -69,7 +69,7 @@ describe SpawnManager do
 			it_should_behave_like "a Rails spawner"
 			it_should_behave_like "a Rails spawner that supports #reload(app_group_name)"
 			it_should_behave_like "a Rails spawner that supports #reload()"
-			include_shared_example_group "a Rails app that lacks RAILS_GEM_VERSION"
+			it_should_behave_like "a Rails app that lacks RAILS_GEM_VERSION"
 		end
 		
 		describe "smart-lv2 spawning" do
@@ -81,7 +81,7 @@ describe SpawnManager do
 			it_should_behave_like "a Rails spawner"
 			it_should_behave_like "a Rails spawner that supports #reload(app_group_name)"
 			it_should_behave_like "a Rails spawner that supports #reload()"
-			include_shared_example_group "a Rails app that lacks RAILS_GEM_VERSION"
+			it_should_behave_like "a Rails app that lacks RAILS_GEM_VERSION"
 		end
 		
 		describe "conservative spawning" do
@@ -92,7 +92,7 @@ describe SpawnManager do
 			it_should_behave_like "a spawner"
 			it_should_behave_like "a Rails spawner"
 			it_should_behave_like "a Rails spawner that supports #reload()"
-			include_shared_example_group "a Rails app that lacks RAILS_GEM_VERSION"
+			it_should_behave_like "a Rails app that lacks RAILS_GEM_VERSION"
 		end
 	end
 	
diff --git a/test/ruby/spec_helper.rb b/test/ruby/spec_helper.rb
index 563b921..376476c 100644
--- a/test/ruby/spec_helper.rb
+++ b/test/ruby/spec_helper.rb
@@ -36,8 +36,8 @@ trap "QUIT" do
 	puts caller
 end
 
-Spec::Runner.configure do |config|
-	config.append_before do
+RSpec.configure do |config|
+	config.before do
 		# Suppress warning messages.
 		PhusionPassenger::DebugLogging.log_level = -1
 		PhusionPassenger::DebugLogging.log_file = nil
@@ -47,7 +47,7 @@ Spec::Runner.configure do |config|
 		PhusionPassenger::Utils.passenger_tmpdir
 	end
 	
-	config.append_after do
+	config.after do
 		tmpdir = PhusionPassenger::Utils.passenger_tmpdir(false)
 		if File.exist?(tmpdir)
 			remove_dir_tree(tmpdir)
