diff -ur passenger-3.0.12.orig/build/documentation.rb passenger-3.0.12/build/documentation.rb
--- passenger-3.0.12.orig/build/documentation.rb	2012-02-03 22:25:41.000000000 +0000
+++ passenger-3.0.12/build/documentation.rb	2012-04-14 14:12:16.335454392 +0100
@@ -21,7 +21,7 @@
 #  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 #  THE SOFTWARE.
 
-ASCIIDOC_FLAGS = "-b html5 -a toc -a theme=flask -a numbered -a toclevels=3 -a icons"
+ASCIIDOC_FLAGS = "-a toc -a theme=flask -a numbered -a toclevels=3 -a icons"
 
 desc "Generate all documentation"
 task :doc => Packaging::ASCII_DOCS
