diff --git a/ext/common/agents/TempDirToucher.c b/ext/common/agents/TempDirToucher.c
index dcf5088..3a82190 100644
--- a/ext/common/agents/TempDirToucher.c
+++ b/ext/common/agents/TempDirToucher.c
@@ -148,7 +148,7 @@ initialize(int argc, char *argv[]) {
 
 static void
 exitHandler(int signo) {
-	write(terminationPipe[1], "x", 1);
+	(void) write(terminationPipe[1], "x", 1);
 }
 
 static void
@@ -180,7 +180,7 @@ maybeDaemonize() {
 		pid = fork();
 		if (pid == 0) {
 			setsid();
-			chdir("/");
+			(void) chdir("/");
 			redirectStdinToNull();
 		} else if (pid == -1) {
 			e = errno;
