diff --git a/ext/common/agents/TempDirToucher.c b/ext/common/agents/TempDirToucher.c
index 3a82190..d3f1b38 100644
--- a/ext/common/agents/TempDirToucher.c
+++ b/ext/common/agents/TempDirToucher.c
@@ -113,7 +113,7 @@ initialize(int argc, char *argv[]) {
 	parseArguments(argc, argv);
 
 	if (logFile != NULL) {
-		fd = open(logFile, O_WRONLY | O_APPEND | O_CREAT);
+		fd = open(logFile, O_WRONLY | O_APPEND | O_CREAT, 0644);
 		if (fd == -1) {
 			e = errno;
 			fprintf(stderr, ERROR_PREFIX
